dimscord/misc.nim

```diff
@@ -174,4 +174,13 @@ proc readPerms*(guild: Guild, member: Member, channel: GuildChannel): PermObj =
         deny = deny or overwrites[m].deny
 
     perms = (perms and deny - deny - deny - 1) or allow
-    result = PermObj(allowed: cast[set[PermEnum]](perms))
\ No newline at end of file
+    result = PermObj(allowed: cast[set[PermEnum]](perms))
+
+func inviteLink*(user: User, permissions: set[PermEnum] = {}, guild = none(Guild), disableGuildSelect = false): string =

```

Comment:
func needs to be a proc for consistency reasons, `user` param should not be User since you are using it for an id especially for guild too and also lines need to be 80-81 characters max for readability.